driver-sync/src/main/com/mongodb/client/gridfs/GridFSDownloadStreamImpl.java

```diff
@@ -156,12 +159,12 @@ public void mark() {
     }
 
     @Override
-    public synchronized void mark(final int readlimit) {
+    public void mark(final int readlimit) {
```

Comment:
This synchronization of mark and reset seems unnecessary, since typical usage of these methods is in a single thread, and also misleading, since currentPosition is used in unsynchronized methods as well.

Reply:
Ack